.pre-commit-config.yaml

```diff
@@ -0,0 +1,50 @@
+# See https://pre-commit.com for more information
+repos:
+  - repo: local
+    hooks:
+      - id: fmt
+        name: fmt
+        description: Format rust files.
+        entry: cargo +nightly fmt
```

Comment:
What is the reason that this isn't the stable toolchain?

Reply:
https://github.com/mozilla/grcov#example-how-to-generate-gcda-files-for-a-rust-project, for the Zprofile needs.

Reply:
@cdata, gives us the crate import granularity, which sadly is not stable yet, but close https://github.com/rust-lang/rustfmt/issues/4991. One of those nightly ones that's worth it for fmt only.